extensions/http-curl/protocols/RESTSender.cpp

```diff
@@ -172,10 +167,6 @@ C2Payload RESTSender::sendPayload(const std::string url, const Direction directi
   }
 }
 
-REGISTER_RESOURCE(RESTSender, "Encapsulates the restful protocol that is built upon C2Protocol.");
+REGISTER_RESOURCE(RESTSender, DescriptionOnly);
```

Comment:
This is a bit of a mess, but it was already like that before this PR:
- `InternalResource` means it inherits from either `Processor` or `ControllerService`, but should not be included in the manifest.  These do not have a `Description` field.
- `DescriptionOnly` means it does not inherit from `Processor` or `ControllerService`, so it will not be included in the manifest, but it does have a `Description` field (which I don't think we use anywhere, but I'm not sure, so I did not want to remove it).

If these `Description`s are in fact not needed, then we could simplify this to:
- if it has a `Description`, then include it in the manifest, detect whether it is a `Processor` or `ControllerService`, and throw a compile error if it's neither;
- if it doesn't have a `Description`, then only register it in the class loader, and skip it in `agent_docs.h` etc completely.

Should we do this?

Reply:
I didn't realize there is a third variable, whether it's included in the manifest. With that in mind, I wouldn't do complete autodetection, but rather keep the explicit declaration of "internalness". Adding an "isInternal" member could work, but leaving it as it is is fine, too.

Reply:
Let's leave it as it is for now; we can revisit this and clean it up later.